src/wp-includes/blocks/post-featured-image.php

```diff
@@ -19,6 +19,12 @@ function render_block_core_post_featured_image( $attributes, $content, $block )
 	}
 	$post_ID = $block->context['postId'];
 
+	// Check is needed for backward compatibility with third-party plugins
+	// that might rely on the `in_the_loop` check; calling `the_post` sets it to true.
+	if ( ! in_the_loop() && have_posts() ) {
+		the_post();
```

Comment:
`the_post` does quite a bit more than setting `in_the_loop` to true, do we really need all that run here?

An alternative would be 
```php
global $wp_query;
$wp_query->in_the_loop = true;
```

Reply:
See my reply on https://github.com/WordPress/gutenberg/pull/45534, which has been merged now.